### PR TITLE
patch for smddp import

### PR DESCRIPTION
**Pull request by carolynwang**
# What does this PR do?

Fixes an `invalid backend` error when starting a HF job with smddp in a method that goes through src/transformers/training_args.py by adding the import statement for smddp, which registers smddp as a torch.distributed backend. 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/training_args.py [hunk 1]
```diff
@@ -1333,6 +1333,8 @@ def _setup_devices(self) -> "torch.device":
             device = torch.device("cuda", local_rank)
             self._n_gpu = 1
         elif is_sagemaker_dp_enabled():
+            import smdistributed.dataparallel.torch.torch_smddp  # noqa: F401
+
```
Comment (philschmid) [inline src/transformers/training_args.py:1337]: i thought this was not needed anymore?

Comment (carolynwang) [reply src/transformers/training_args.py:1337]: this torch_smddp import is required to register smddp as a PT backend - the older version of smddp required a [different] import as well as other points of code changes which is no longer needed

Comment (philschmid) [reply src/transformers/training_args.py:1337]: I see i guess this slipped through here: https://github.com/huggingface/transformers/pull/16371/files#diff-bfceaff300c851b8e24fc50dc6638482abaec8f7d2a718e877c3828c166bcf79

**Review by sgugger — APPROVED** [review]
Thanks for fixing!

**Pull request merged**